politeiawww/proposals.go

```diff
@@ -890,6 +887,69 @@ func (p *politeiawww) processProposalDetails(propDetails www.ProposalsDetails, u
 	return &reply, nil
 }
 
+// voteSummaries fetches the voting summary information for a set of
+// proposals.
+func (p *politeiawww) voteSummaries(tokens []string, bestBlock uint64) (map[string]www.VoteSummary, error) {
+
+	r, err := p.decredBatchVoteSummary(tokens)
+	if err != nil {
+		return nil, err
+	}
+
+	voteSummaries := make(map[string]www.VoteSummary)
+
+	for token, summary := range r.Summaries {
+		results := convertVoteOptionResultsFromDecred(summary.Results)
+
+		endHeight, err := strconv.ParseUint(summary.EndHeight, 10, 64)
+		if err != nil {
+			endHeight = 0
+		}
+
+		vs := www.VoteSummary{
+			Status:           voteStatusFromVoteSummary(summary, bestBlock),
+			EligibleTickets:  uint32(summary.EligibleTicketCount),
+			EndHeight:        endHeight,
+			BestBlock:        bestBlock,
+			QuorumPercentage: summary.QuorumPercentage,
+			PassPercentage:   summary.PassPercentage,
+			Results:          results,
+		}
+
+		voteSummaries[token] = vs
+	}
+
+	return voteSummaries, nil
+}
+
+// processBatchVoteSummary the voting summary information for a set of
+// proposals.
+func (p *politeiawww) processBatchVoteSummary(batchVoteSummary www.BatchVoteSummary) (*www.BatchVoteSummaryReply, error) {
+	log.Tracef("processBatchVoteSummary")
+
+	bb, err := p.getBestBlock()
+	if err != nil {
+		return nil, err
+	}
+
+	summaries, err := p.voteSummaries(batchVoteSummary.Tokens, bb)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(summaries) != len(batchVoteSummary.Tokens) {
+		return nil, www.UserError{
```

Comment:
A user error contains a `ErrorContext` field that allows you to give context to the error message when appropriate.  It would make sense here to use the `ErrorContext` to specify which tokens were not found.